Index: vagalume/src/lastfm-ws.c
===================================================================
--- vagalume.orig/src/lastfm-ws.c	2010-05-04 13:42:03.000000000 +0100
+++ vagalume/src/lastfm-ws.c	2010-05-04 13:42:24.000000000 +0100
@@ -751,7 +751,7 @@ lastfm_ws_get_track_tags                
 
         lastfm_ws_http_request (session->srv, method,
                                 HTTP_REQUEST_GET, FALSE, &doc, &node,
-                                "artist", track->artist,
+                                "artist", artist,
                                 extraparam, extraparamvalue,
                                 NULL);
 
@@ -811,7 +811,7 @@ lastfm_ws_add_tags                      
 
         lastfm_ws_http_request (session->srv, method, HTTP_REQUEST_POST,
                                 TRUE, &doc, &node,
-                                "artist", track->artist,
+                                "artist", artist,
                                 "sk", session->key,
                                 "tags", tags,
                                 extraparam, extraparamvalue,
@@ -863,7 +863,7 @@ lastfm_ws_remove_tag                    
 
         lastfm_ws_http_request (session->srv, method, HTTP_REQUEST_POST,
                                 TRUE, &doc, &node,
-                                "artist", track->artist,
+                                "artist", artist,
                                 "sk", session->key,
                                 "tag", tag,
                                 extraparam, extraparamvalue,
